### PR TITLE
cleanup(storage): remove some uses of `CurlClient`

### PR DESCRIPTION
**Pull request by coryan**
Remove uses of `CurlClient`. One lingering in the implementation of
`RestClient` itself.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9759)

### CODE REVIEW

File: google/cloud/storage/grpc_plugin_test.cc [hunk 1]
```diff
@@ -83,14 +74,11 @@ TEST(GrpcPluginTest, UnsetConfigCreatesCurl) {
   auto const* const retry =
       dynamic_cast<RetryClient*>(ClientImplDetails::GetRawClient(client).get());
   ASSERT_THAT(retry, NotNull());
-  auto const* const curl = dynamic_cast<CurlClient*>(retry->client().get());
+  auto const* const curl = dynamic_cast<RestClient*>(retry->client().get());
```
Comment (devbww) [inline google/cloud/storage/grpc_plugin_test.cc:77]: Should this still be called `curl`?

File: google/cloud/storage/grpc_plugin_test.cc [hunk 2]
```diff
@@ -100,14 +88,11 @@ TEST(GrpcPluginTest, NoneConfigCreatesCurl) {
...
```
Comment (devbww) [inline google/cloud/storage/grpc_plugin_test.cc:91]: Should this still be called `curl`?

File: google/cloud/storage/internal/hybrid_client.cc [hunk 3]
```diff
@@ -27,7 +29,7 @@ std::shared_ptr<RawClient> HybridClient::Create(Options const& options) {
 
 HybridClient::HybridClient(Options const& options)
     : grpc_(GrpcClient::Create(DefaultOptionsGrpc(options))),
-      curl_(CurlClient::Create(DefaultOptionsWithCredentials(options))) {}
+      curl_(RestClient::Create(DefaultOptionsWithCredentials(options))) {}
```
Comment (devbww) [inline google/cloud/storage/internal/hybrid_client.cc:32]: Should this still be called `curl_`?

Comment (coryan) [reply google/cloud/storage/internal/hybrid_client.cc]: Fixed.

File: google/cloud/storage/grpc_plugin_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/grpc_plugin_test.cc]: Fixed

File: google/cloud/storage/grpc_plugin_test.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/grpc_plugin_test.cc]: Fixed

**Review by devbww — APPROVED** [review]

**Pull request merged**